modules/order/src/Entity/Order.php

```diff
@@ -87,12 +88,28 @@ public function preSave(EntityStorageInterface $storage) {
     }
 
     // Recalculate the total.
+    $this->calculateTotal();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function calculateTotal() {
     // @todo Rework this once pricing is finished.
     $this->total_price->amount = 0;
     foreach ($this->getLineItems() as $line_item) {
       $this->total_price->amount += $line_item->total_price->amount;
       $this->total_price->currency_code = $line_item->total_price->currency_code;
     }
+
+    /** @var \Drupal\commerce_order\Plugin\Field\FieldType\AdjustmentItem $adjustment */
+    // @todo FieldItemList implementation which just gets Adjustment that
+    // returns Adjustment objects directly.
+    foreach ($this->adjustments as $adjustment) {
+      $adjustment_price = $adjustment->toPrice();
+      // @todo The price field should have an ::addPrice method.
```

Comment:
What are thoughts on adding a method to the price field that consumes a price object and adds its to the amount value (if currency codes match.)